android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/Utils.java

```diff
@@ -1119,4 +1126,64 @@ public PwsResultBuilder setPwsTripTimeMillis(PwsResult pwsResult, long timeMilli
       return this;
     }
   }
+
+  /**
+   * Determines if a file is gzipped by examining the signature of
+   * the file, which is the first two bytes.
+   * @param file to be determined if is gzipped.
+   * @return true If the contents of the file are gzipped otherwise false.
+   */
+  public static boolean isGzippedFile(File file) {
+    InputStream input;
+
+    try {
+      input = new FileInputStream(file);
+    } catch(FileNotFoundException e) {
+      return false;
+    }
+
+    PushbackInputStream pb = new PushbackInputStream(input, GZIP_SIGNATURE_LENGTH);
```

Comment:
Why PushbackInputStream? Could you use FileInputStream.read instead? It doesn't seem like we need the pushback functionality.